src/aws_encryption_sdk/keyring/raw_keyring.py

```diff
@@ -0,0 +1,390 @@
+            ):
+                # Wrapped EncryptedDataKey to deserialized EncryptedData
+                encrypted_wrapped_key = deserialize_wrapped_key(
+                    wrapping_algorithm=self._wrapping_algorithm,
+                    wrapping_key_id=self.key_name,
+                    wrapped_encrypted_key=key,
+                )
+                # EncryptedData to raw key string
+                try:
+                    plaintext_data_key = self._wrapping_key_structure.decrypt(
+                        encrypted_wrapped_data_key=encrypted_wrapped_key,
+                        encryption_context=decryption_materials.encryption_context,
+                    )
+
+                except Exception as error:  # pylint: disable=broad-except
+                    logger = logging.getLogger()
+                    logger.error(error.__class__.__name__, ":", str(error))
+                    return decryption_materials
+
+                if plaintext_data_key:
+                    # Create a keyring trace
+                    keyring_trace = KeyringTrace(
+                        wrapping_key=self._key_provider, flags={KeyringTraceFlag.WRAPPING_KEY_DECRYPTED_DATA_KEY}
+                    )
+
+                    # Update decryption materials
+                    data_encryption_key = RawDataKey(
+                        key_provider=MasterKeyInfo(provider_id=self._key_provider.provider_id, key_info=self.key_name),
+                        data_key=plaintext_data_key,
+                    )
+                    decryption_materials.add_data_encryption_key(data_encryption_key, keyring_trace)
+
+                if decryption_materials.data_encryption_key:
+                    return decryption_materials
+
+        return decryption_materials
+
+
+@attr.s
+class RawRSAKeyring(Keyring):
+    """Public class for Raw RSA Keyring.
+
+    :param str key_namespace: String defining the keyring ID
+    :param bytes key_name: Key ID
+    :param wrapping_key: Encryption key with which to wrap plaintext data key
+    :type wrapping_key: object
+    :param wrapping_algorithm: Wrapping Algorithm with which to wrap plaintext data key
+    :type wrapping_algorithm: WrappingAlgorithm
+    :param key_provider: Complete information about the key in the keyring
+    :type key_provider: MasterKeyInfo
+    """
+
+    key_namespace = attr.ib(validator=attr.validators.instance_of(six.string_types))
+    key_name = attr.ib(validator=attr.validators.instance_of(six.binary_type))
+    # _wrapping_key = attr.ib(repr=False, validator=attr.validators.in_([rsa.RSAPrivateKey, rsa.RSAPublicKey]))
+    # ----- need to figure out how to do this correctly
+    _wrapping_key = attr.ib(repr=False, validator=attr.validators.instance_of(object))
+    _wrapping_algorithm = attr.ib(repr=False, validator=attr.validators.instance_of(WrappingAlgorithm))
+
+    @classmethod
+    def fromPEMEncoding(cls):
+        key_namespace = attr.ib(validator=attr.validators.instance_of(six.string_types))
```

Comment:
Unfortunately, attrs doesn't work like this on classmethods. We'll need to take inputs like any other method.